pulsar-common/src/main/java/org/apache/pulsar/common/util/netty/EventLoopUtil.java

```diff
@@ -0,0 +1,78 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.common.util.netty;
+
+import java.util.concurrent.ThreadFactory;
+
+import io.netty.bootstrap.ServerBootstrap;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.epoll.Epoll;
+import io.netty.channel.epoll.EpollChannelOption;
+import io.netty.channel.epoll.EpollEventLoopGroup;
+import io.netty.channel.epoll.EpollMode;
+import io.netty.channel.epoll.EpollServerSocketChannel;
+import io.netty.channel.epoll.EpollSocketChannel;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.channel.socket.ServerSocketChannel;
+import io.netty.channel.socket.SocketChannel;
+import io.netty.channel.socket.nio.NioServerSocketChannel;
+import io.netty.channel.socket.nio.NioSocketChannel;
+
+public class EventLoopUtil {
+
+    /**
+     * @return an EventLoopGroup suitable for the current platform
+     */
+    public static EventLoopGroup newEventLoopGroup(int nThreads, ThreadFactory threadFactory) {
+        if (Epoll.isAvailable()) {
+            return new EpollEventLoopGroup(nThreads, threadFactory);
+        } else {
+            // Fallback to NIO
+            return new NioEventLoopGroup(nThreads, threadFactory);
+        }
+    }
+
+    /**
+     * Return a SocketChannel class suitable for the given EventLoopGroup implementation
+     *
+     * @param eventLoopGroup
+     * @return
+     */
+    public static Class<? extends SocketChannel> getClientSocketChannelClass(EventLoopGroup eventLoopGroup) {
+        if (eventLoopGroup instanceof EpollEventLoopGroup) {
```

Comment:
can we still add check `Epoll.isAvailable()` to make sure os supports the group?